scripts/upgrade

```diff
@@ -1,81 +1,156 @@
+# CHECK THE PATH
+#=================================================
+
+# Normalize the URL path syntax
+path_url=$(ynh_normalize_url_path $path_url)
+
+#=================================================
+# STANDARD UPGRADE STEPS
+#=================================================
+# DOWNLOAD, CHECK AND UNPACK SOURCE
+#=================================================
+
+# Download, check integrity, uncompress and patch the source from app.src
+ynh_setup_source "$final_path"
+
+#=================================================
+# NGINX CONFIGURATION
+#=================================================
+
+# Create a dedicated nginx config
+ynh_add_nginx_config
+
+#=================================================
+# CREATE DEDICATED USER
+#=================================================
+
+# Create a system user
 ynh_system_user_create $app
 
-# Modify dokuwiki conf
-sed -i "s@YNH_ADMIN_USER@$admin@g" ../conf/dokuwiki.php
+#=================================================
+# PHP-FPM CONFIGURATION
+#=================================================
+
+# Create a dedicated php-fpm config
+ynh_add_fpm_config
+
+#=================================================
+# SPECIFIC UPGRADE
+#=================================================
 
-# Copy files to the right place
-final_path=/var/www/$app
-sudo mkdir -p $final_path
+# TODO Taken from old "upgrade" script. Should check if it is needed and what it does
+if [ -d "${final_path}/data/media" ]; then
+	chown -R $app:root $final_path/{data/attic,data/cache,data/index,data/locks,data/media*,data/meta,data/pages,data/tmp}
+fi
+
+# Set the "language" of Dokuwiki
+# Might overwrite user choice if language has been change directly in Dokuwiki admin panel
+# Should be enhance to avoid this. For now, it is easier to overwrite as there is no need to take care of existing configuration
+ynh_replace_string "__YNH_LANGUAGE__" "$language" "../conf/local.php"
 
-# Get source
-SETUP_SOURCE
+# Set the "admin" user of Dokuwiki
+ynh_replace_string "__YNH_ADMIN_USER__" "$admin" "../conf/local.php"
 
-sudo cp ../conf/dokuwiki.php $final_path/conf
+# Copy Yunohost specific configuration
+cp ../conf/local.php $final_path/conf
 
 # Do not override ACL configuration file
 if [ ! -f "$final_path/conf/acl.auth.php" ]; then
-	sudo cp ../conf/acl.auth.php $final_path/conf
+	cp ../conf/acl.auth.php $final_path/conf
 fi
 
 # Remove upgrade notification
 # See https://www.dokuwiki.org/update_check
-sudo touch $final_path/doku.php
+touch $final_path/doku.php
 
-# Remove deleted files
+# Remove files not used anymore after upgrade
 # See https://www.dokuwiki.org/install:unused_files
-if [ -f "../sources/data/deleted.files" ]; then
-	grep -Ev '^($|#)' ../sources/data/deleted.files | xargs -I {} sudo rm -vrf $final_path/{}
-fi
+if [ -f "$final_path/data/deleted.files" ]; then
+	# Move to the dokuwiki installation folder so the "official" commands can be used without adaptation
+	cd $final_path
```

Comment:
Using a sub process would be more efficient here.

Reply:
What is "sub process" ?
After thinking, this test is not needed as the "deleted.files" will always be installed the "dokuwiki sources"

Reply:
What I would say is, instead of using cd $final_path and cd -, it could be easier to use a sub process.
```
( cd $final_path
grep -Ev '^($|#)' data/deleted.files | xargs -n 1 rm -fr )
```
Using brackets will allow to use only one `cd`, because the brackets will start a new shell, so all the commands will run in this other shell.
Using brackets is a good solution to not really move from the current path in your script.

Reply:
I didn't know about it! Seems more efficient but less easy to read and to understand. I will add a comment this your explanation!

While coding, I preferred to keep the solution proposed on the Dokuwiki website instead of trying to understand the one in the script.

Reply:
Don't hesitate to add a comment to explain why and how. It's always better to have comments to explain.